File: leetcode/0001-1000/0724-find-pivot-index/FindPivotIndex.java
public class FindPivotIndex {
  public static void main(String[] args) {
    // int[] numbers = new int[] { 1, 7, 3, 6, 5, 6 };
    int[] numbers = new int[] { -1, 1, 2 };

    System.out.println(new Solution().pivotIndex(numbers));
  }
}

class Solution {
  public int pivotIndex(int[] numbers) {
    int[] leftAccumulated = new int[numbers.length + 1];
    int[] rightAccumulated = new int[numbers.length + 1];
    int length = numbers.length;

    for (int i = 0; i < length; i++) {
      leftAccumulated[i + 1] = leftAccumulated[i] + numbers[i];
      rightAccumulated[length - i - 1] = rightAccumulated[length - i] + numbers[length - i - 1];
    }

    for (int i = 0; i < length; i++) {
      if (leftAccumulated[i] == rightAccumulated[i + 1]) {
        return i;
      }
    }

    return -1;
  }
}

File: leetcode/0001-1000/0724-find-pivot-index/FindPivotIndex.ts
function pivotIndex(numbers: number[]): number {
  if (numbers.length < 2) {
    return 0;
  }

  const accumulated = new Array(numbers.length + 1).fill(0);

  numbers.forEach((v, i) => {
    accumulated[i + 1] = accumulated[i] + numbers[i]
  });

  for (let i = 0; i < numbers.length; i++) {
    const left = accumulated[i] - accumulated[0];
    const right = accumulated[numbers.length] - accumulated[i + 1];

    if (left === right) {
      return i;
    }
  }

  return -1;
};

console.log(pivotIndex([1, 7, 3, 6, 5, 6]))
console.log(pivotIndex([2, 1, -1]));

